From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Denery <dorofeevij@gmail.com>
Date: Mon, 31 Oct 2022 14:20:52 +0300
Subject: [PATCH] 
 Fix-EntityCombustEvent-cancellation-cant-fully-prevent-entities-from-being-set-on-fire


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index f2908cf61ff28bef44fcf46b15cf585e942fd7ce..148034a8fc4167ff0e77a8eb1dd205c38a91d548 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -3137,6 +3137,11 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
             if (!entityCombustEvent.isCancelled()) {
                 this.setSecondsOnFire(entityCombustEvent.getDuration(), false);
             }
+            // Paper start - fix EntityCombustEvent cancellation can't prevent entities from being set on fire
+            else {
+                this.setRemainingFireTicks(this.remainingFireTicks - 1);
+            }
+            // Paper end
             // CraftBukkit end
         }
 
